Desktop: Dev branch not working with node 16

### Operating system

Windows

### Joplin version

last version from dev branch 

### Desktop version info

_No response_

### Current behaviour

Steps to reproduce:

- Before installing Node, I deleted everything related to nvm, node, yarn, and npm. This means uninstalling Node from Control Panel and deleting everything related to them inside %APPDATA% (including caching)
- I installed Node 16.20.2 from [here](https://nodejs.org/en/download)
- I followed the steps from your BUILD.md (I did not open Visual Studio Code at all since, every command was ran with Windows Powershell as Administrator )
- After running `yarn install`, this was the output:

```
➤ YN0008: │ husky@patch:husky@npm%3A3.1.0#./.yarn/patches/husky-npm-3.1.0-5cc13e4e34.patch::version=3.1.0&hash=d9603f&locator=root%40workspace%3A. must be rebuilt because its dependency tree changed
➤ YN0008: │ keytar@npm:7.9.0 must be rebuilt because its dependency tree changed
➤ YN0008: │ sharp@npm:0.33.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ sqlite3@npm:5.1.6 [ea905] must be rebuilt because its dependency tree changed
➤ YN0008: │ canvas@npm:2.11.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ tesseract.js@npm:5.0.4 must be rebuilt because its dependency tree changed
➤ YN0008: │ core-js@npm:3.33.1 must be rebuilt because its dependency tree changed
➤ YN0008: │ core-js-pure@npm:3.33.1 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-free@npm:5.15.4 must be rebuilt because its dependency tree changed
➤ YN0008: │ electron@npm:29.1.0 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-common-types@npm:0.2.36 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-common-types@npm:6.4.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-common-types@npm:6.1.2 must be rebuilt because its dependency tree changed
➤ YN0007: │ root@workspace:. must be built because it never has been before or the last one failed
➤ YN0008: │ @joplin/app-mobile@workspace:packages/app-mobile must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-svg-core@npm:1.2.36 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-regular-svg-icons@npm:5.15.4 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-solid-svg-icons@npm:5.15.4 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-svg-core@npm:6.4.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-brands-svg-icons@npm:6.4.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-regular-svg-icons@npm:6.4.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-solid-svg-icons@npm:6.4.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/fontawesome-svg-core@npm:6.1.2 must be rebuilt because its dependency tree changed
➤ YN0008: │ @fortawesome/free-solid-svg-icons@npm:6.1.2 must be rebuilt because its dependency tree changed
➤ YN0009: │ root@workspace:. couldn't be built successfully (exit code 1, logs can be found here: <path_to_build_log>\build.log)
```

The last line that starts with root@workspace is problematic. I also attached the logs, and there you can find the following errors:
```
➤ YN0000: [@joplin/app-desktop]: Error: Cannot find module 'readline/promises'
[[90m00:53:17[39m] Error: Command failed with exit code 1: yarn run buildParallel

```


### Expected behaviour

Regarding the first error: I searched on the Node website, Readline section, the correct syntax:

- the correct syntax for Node 16 is [this one](https://nodejs.org/docs/latest-v16.x/api/readline.html#readline)
- the correct syntax for Node 17 and above is [this one](https://nodejs.org/docs/latest-v17.x/api/readline.html#readline)

**Fix**: 

- Either update the documentation [here](https://github.com/laurent22/joplin/blob/dev/readme/dev/BUILD.md#required-dependencies) from version 16 to 17
- Or change inside joplin>packages>default-plugins>utils>waitForCliInput.ts the first line from `const readline = require('readline/promises');` to `const readline = require('readline');`

After this change, I was able to start the application. The second error was because of @joplin/lib, but I do not know the solution.

### Logs

[error node 16.txt](https://github.com/laurent22/joplin/files/14812473/error.node.16.txt)

--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "url": "git+https://github.com/laurent22/joplin.git"
   },
   "engines": {
-    "node": ">=16"
+    "node": ">=18"
   },
   "scripts": {
     "buildApiDoc": "yarn workspace joplin start apidoc ../../readme/api/references/rest_api.md",

--- a/readme/dev/BUILD.md
+++ b/readme/dev/BUILD.md
@@ -18,7 +18,7 @@ There are also a few forks of existing packages under the "fork-*" name.
 
 ## Required dependencies
 
-- Install Node 16+. On Windows, also install the build tools - https://nodejs.org/en/
+- Install Node 18+. On Windows, also install the build tools - https://nodejs.org/en/
   - [Enable Yarn](https://yarnpkg.com/getting-started/install): `corepack enable`
 - macOS: Install Cocoapods - `brew install cocoapods`. Apple Silicon [may require libvips](https://github.com/laurent22/joplin/pull/5966#issuecomment-1007158597) - `brew install vips`.
 - Linux: Install dependencies - `sudo apt install build-essential libnss3 libsecret-1-dev python rsync libgbm-dev libatk-bridge2.0-0 libgtk-3.0 libasound2`

